src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,29 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("User cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException("User with this login already exists");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserDataException("Login must be at least 6 characters long");
```

Comment:
It would be better to move the hardcoded value `6` to a constant field. It will make the code more maintainable. If you need to change this value in the future, you can do it in one place.